qadence/constructors/feature_maps.py

```diff
@@ -76,21 +82,28 @@ def fm_parameter(
     else:
         scaled_fparam = scaling * fparam + shift
 
+    return scaled_fparam
+
+
+def fm_parameter_func(fm_type: BasisSet | Callable | str) -> type[Function]:
+    def ident_fn(x: TParameter) -> TParameter:
+        return x
+
     # Transform feature parameter
     if fm_type == BasisSet.FOURIER:
-        transformed_feature = scaled_fparam
+        transform_func = ident_fn
     elif fm_type == BasisSet.CHEBYSHEV:
-        transformed_feature = acos(scaled_fparam)
-    elif inspect.isclass(fm_type) and issubclass(fm_type, Function):
-        transformed_feature = fm_type(scaled_fparam)
+        transform_func = acos
+    elif callable(fm_type):
+        transform_func = fm_type
     else:
         raise NotImplementedError(
             f"Feature map type {fm_type} not implemented. Choose an item from the BasisSet "
-            f"enum: {[bs.name for bs in BasisSet]}, or your own sympy.Function to wrap "
+            f"enum: {[bs.name for bs in BasisSet]}, or your own sympy function to wrap "
             "the given feature parameter with."
         )
 
-    return transformed_feature
+    return transform_func  # type: ignore [return-value]
```

Comment:
With the return type suggestion, could the ignore be removed ?

Reply:
Yes indeed.